players/management/commands/generate_players.py

```diff
@@ -0,0 +1,98 @@
+import argparse
+import os
+import uuid
+
+from django.core.management import BaseCommand
+from django.test import RequestFactory
+from importlib import import_module
+from django.conf import settings
+from django.contrib.auth.models import User
+from players.views import code as code_view
+
+
+_AVATAR_CODES_DIRECTORY = 'players/avatar_examples'
+
+# Code file listing
+
+
+def _strip_prefix(prefix, string):
+    if string.startswith(prefix):
+        return string[len(prefix):]
+
+
+def _get_available_code_files(base_directory):
+    for dirpath, dirnames, filenames in os.walk(base_directory):
+        for f in filenames:
+            if not f.startswith('_') and f.endswith('.py'):
+                parent_dir = _strip_prefix(base_directory, dirpath)
+                yield os.path.join(parent_dir, f)
+
+_AVATAR_CODES = list(_get_available_code_files(_AVATAR_CODES_DIRECTORY))
+
+# Code file loading
+
+
+def _load_code_file(filename):
+    if not filename.endswith('.py'):
+        filename = filename + '.py'
+
+    filepath = os.path.join(_AVATAR_CODES_DIRECTORY, filename)
+
+    with open(filepath) as f:
+        return f.read()
+
+
+class LoadCodeAction(argparse.Action):
+
+    def __init__(self, option_strings, dest, **kwargs):
+        super(LoadCodeAction, self).__init__(option_strings, dest, **kwargs)
+
+    def __call__(self, parser, namespace, values, option_string):
+        values = _load_code_file(values)
+        setattr(namespace, self.dest, values)
+
+
+class Command(BaseCommand):
+    # Show this when the user types help
+    help = "Generate users for the game"
+
+    def __init__(self, *args, **kwargs):
+        super(Command, self).__init__(*args, **kwargs)
+        self.engine = import_module(settings.SESSION_ENGINE)
+        self.request_factory = RequestFactory()
+
+    def add_arguments(self, parser):
+        parser.add_argument('--num-users', type=int, required=True,
+                            help='Number of users to create')
+        parser.add_argument('--avatar-code', choices=_AVATAR_CODES,
+                            action=LoadCodeAction, required=True,
+                            help='The code to use for the avatar.')
```

Comment:
I don't like the naming of the options here: one is users, the other is avatar, the command is players. I can't remember which is which!
Also, do we want required named rather than positional requirements ( `generate_players 3 dummy_avatar.py` vs `generate_players --num-users=3 --avatar-code=dummy_avatar.py`)